src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/FormattingVisitor.cs

```diff
@@ -188,12 +188,12 @@ public override void VisitMarkupEndTag(MarkupEndTagSyntax node)
 
         public override void VisitMarkupTagHelperElement(MarkupTagHelperElementSyntax node)
         {
-            var isComponent = IsComponentTagHelperNode(node);
+            var causesIndentation = CausesGeneratedCSharpIndentation(node);
```

Comment:
Very succinct change!

At some point I agree with your offline view that we should probably decouple the codegen implications on the SyntaxTree but I like how this is mostly simple for now.